src/inmanta/server/services/orchestrationservice.py

```diff
@@ -1032,87 +1032,96 @@ async def release_version(
     ) -> Apireturn:
-        model = await data.ConfigurationModel.get_version(env.id, version_id, connection=connection)
-        if model is None:
-            return 404, {"message": "The request version does not exist."}
-
-        # Already mark undeployable resources as deployed to create a better UX (change the version counters)
-        undep = await model.get_undeployable()
-        now = datetime.datetime.now().astimezone()
-
-        if undep:
-            undep_ids = [ResourceVersionIdStr(rid + ",v=%s" % version_id) for rid in undep]
-            # not checking error conditions
-            await self.resource_service.resource_action_update(
-                env,
-                undep_ids,
-                action_id=uuid.uuid4(),
-                started=now,
-                finished=now,
-                status=const.ResourceState.undefined,
-                action=const.ResourceAction.deploy,
-                changes={},
-                messages=[],
-                change=const.Change.nochange,
-                send_events=False,
-                connection=connection,
-            )
-
-            skippable = await model.get_skipped_for_undeployable()
-            if skippable:
-                skippable_ids = [ResourceVersionIdStr(rid + ",v=%s" % version_id) for rid in skippable]
-                # not checking error conditions
-                await self.resource_service.resource_action_update(
-                    env,
-                    skippable_ids,
-                    action_id=uuid.uuid4(),
-                    started=now,
-                    finished=now,
-                    status=const.ResourceState.skipped_for_undefined,
-                    action=const.ResourceAction.deploy,
-                    changes={},
-                    messages=[],
-                    change=const.Change.nochange,
-                    send_events=False,
-                    connection=connection,
-                )
+        async with data.ConfigurationModel.get_connection(connection) as connection:
+            async with connection.transaction():
+                model = await data.ConfigurationModel.get_version(env.id, version_id, connection=connection)
+                if model is None:
+                    return 404, {"message": "The request version does not exist."}
 
-        increments: tuple[abc.Set[ResourceIdStr], abc.Set[ResourceIdStr]] = await self.resource_service.get_increment(
-            env, version_id
-        )
+                if model.released:
+                    raise Conflict(f"The version {version_id} on environment {env} is already released.")
 
-        increment_ids, neg_increment = increments
-        await self.resource_service.mark_deployed(env, neg_increment, now, version_id)
+                # Already mark undeployable resources as deployed to create a better UX (change the version counters)
+                undep = await model.get_undeployable()
```

Comment:
Another query on a separate connection

Reply:
not a query strangely enough, I could drop the async keyword on them, if you think that would clarify it.

Reply:
[It is](https://github.com/inmanta/inmanta-core/blob/8f0620df3c673284fcd0a0211b6047900a23d623/src/inmanta/data/__init__.py#L4510), isn't it? Am I missing something?

Reply:
I think it is this one: https://github.com/inmanta/inmanta-core/blob/8f0620df3c673284fcd0a0211b6047900a23d623/src/inmanta/data/__init__.py#L5404

Reply:
Wow, didn't even realize I was looking in the wrong class...

Reply:
do I remove the async?

Reply:
I see no reason not to (unless we expose this somewhere but I don't think we do)